### PR TITLE
Fix various issues

### PR DESCRIPTION
**Pull request by jonparrott**
* Update to latest gcloud
* Fix logging setup
* Retry on transient errors
* Deal with conflicts on topic and subscription creation.

Other stuff:
* Switch to py.test
* Created coveragerc

### CODE REVIEW

File: psq/queue.py [hunk 1]
```diff
@@ -64,7 +69,11 @@ def _get_or_create_subscription(self):
         if not subscription.exists():
```
Comment (waprin) [inline psq/queue.py]: not sure, but it seems if you're wrapping in a try catch there's not a huge purpose to the first if. As Danny would say IBTAFTP . Plus you save the round trip of the exists check.

Comment (jonparrott) [inline psq/queue.py]: So this is sort of a compromise here, it covers all cases where something could go wrong.

Comment (waprin) [inline psq/queue.py]: In what case does the if statement protect you that the try/catch doesnt?

Comment (jonparrott) [inline psq/queue.py]: That's fair, but I'd rather pay the one-time cost of a check (this only happens at startup) than depend on the API catching the conflict. It's a bit more defensive.

**Comment by waprin** [comment]
lgtm

File: psq/queue.py (hunk 1 above)
Comment (waprin) [inline psq/queue.py]: alright, shruggie

**Pull request merged**